tests/warn/i20951.scala

```diff
@@ -0,0 +1,20 @@
+//> using options  -Wunused:params
```

Comment:
If it's the case that this subsumes the ticket for warning in a for comprehension (e.g. the case `for x <- list yield y // warn for x`, #18289) then we should add a test for that syntax as well since that ticket will no longer require any changes.

Reply:
I also would not consider it a problem that the for comprehension warning is unused parameter instead of local. The only way around that would run into the same issues as #18854 and may be subject to the for-comprehension desugaring changes.